### ISSUE TITLE
Cumulative budget forecast gives incorrect percentages when assets are included

### ISSUE THREAD

**Issue by xnyhps** [simonmichael/hledger#862]
I just ran into the following weird bug. Simplified example:

```
P 2018/01/26 SHARE €10

2018/05/17 Dividend
    revenue:dividend       €-10
    assets:pension          €10

2018/05/17 Buy
    assets:pension:shares   1 SHARE @ €10
    assets:pension          €-10

~ monthly from 2018/06 to 2018/07
    assets:pension                    €1
    assets:bank
```

Gives the following forecast:

```
hledger -f bug.journal bal -M --budget --cumulative --forecast --value
Budget performance in 2018/05/01-2018/06/30:

                ||                      May                        Jun
================++=====================================================
 <unbudgeted>   ||    €-10                       €-10
 assets:bank    ||       0                        €-1 [ 100% of   €-1]
 assets:pension ||     €10                       €11 [10100% of    €1]
----------------++-----------------------------------------------------
                ||       0                          0 [             0]
```

10100% of €1 is €101, not €11, so this can't be right. It looks like it might accidentally multiply the asset by the asset price twice maybe?

**Comment by simonmichael** [comment]
Thanks for the report. --cumulative and --budget haven't been tested before.

**Comment by simonmichael** [comment]
PS in fact I have never used --cumulative for anything. What made you use that flag ?

**Comment by xnyhps** [comment]
Sorry, I don't follow. [`--cumulative` and `--budget`](https://github.com/simonmichael/hledger/wiki/Budgeting-and-forecasting#envelope-budget) seems to be recommended by the wiki for envelope budgets. I was following along that guide.

I did incorrectly think I had to combine `--budget` and `--forecast`, but that wasn't what the guide said. It does make sense to me to show the forecast total next to the budgeted total, though.
